docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,220 @@
+5. Data formatting and handling by the event streaming service is out of the scope of this epic. The event streaming service should be able to process the data from all versions of the on-premises environment.
+
+
+## Proposal
+
+### Data Sending Flow
+
+All of the event data for this cluster will be queried from the DB, packaged, and sent to the data hosting endpoint.
+
+Data will be sent when:
+1. A cluster installation attempt finishes (regardless of success or failure)
+
+Before the data is sent, Assisted Service will get a remote configuration from an endpoint we specify.
+
+Sending data should not interfere with customer workloads and deployment
+
+### Assisted-Service Changes 
+
+Changes that need to be made in the `Assisted-Service` to support this:
+- Needs to know what the data hosting service endpoint is and how to authenticate into it
+- Needs code changes when an install is attempted to query and send data
+- Needs to know how to format the data to send to the data hosting service
+    - Includes sanitizing the data such as hashing user's emails like in the [event scraper](https://github.com/openshift-assisted/assisted-events-scrape/blob/master/assisted-events-scrape/utils/anonymizer.py#L25)
+- Needs to expose an option for users to opt out of this feature
+- Needs to know where and how to get the remote configuration 
+
+
+### Additional Changes
+
+- Documentation about how to opt out
+- Documentation of requirements: non-disconnected environment or local s3 storage?
+- Host a remote configuration (using a CDN)
+
+### Remote Configuration
+
+The remote configuration will specify configuration data for the assisted-service such as `DATA_COLLECTION_URL`. This can be cached in a CDN.
+
+The assisted service will get this remote configuration:
+
+1. When the assisted service initially starts
+2. Every time the assisted service will send data it'll get this configuration first
+
+If the assisted service can't get the latest configuration, it'll fallback to the last configuration received or the hard-coded default it was deployed with.
+
+#### Benefits
+1. Future-proof this feature
+    - The configuration is not reliant on the version of the assisted-service deployed
+    - E.g. if the data collection endpoint changes, older versions will be able to send it to the new endpoint without having to update their client
+2. Allows flexibility for our team
+    - We can adjust the configurations on the fly such as where data is sent to
+    - E.g. if the original data collection endpoint is unavailable or if we choose to use a different service later on, we only need to change the configuration endpoint 
+
+#### Drawbacks
+1. Another deployment artifact to update and maintain
+2. Doesn't fully future-proof against the remote configuration's URL changes
```

Comment:
This is my point above. I don't think it makes sense to solve this for the real endpoint when we'd have the same "problem" for the meta-endpoint.

Reply:
Agreed, but using it as a general configuration with other toggles (as you mentioned in another comment) sounds useful!